osi_environment.proto

```diff
@@ -324,4 +336,156 @@ message EnvironmentalConditions
         //
         optional uint32 seconds_since_midnight = 1;
     }
+
+    //
+    // \brief Specification of the cloud layer.
+    //
+    // \note within this message there is the cloud state level.
+    //
+    message CloudLayer
+    {
+        // Definition of the cloud state level, i.e. cloud state and sky visualization settings.
+        //
+        optional CloudLayerLevel cloud_layer_level = 1;
+
+        // Defines the cloud state levels given by observation of total cloud amount in oktas.
+        //
+        // For visual alignment please call the reference[1]
+        //
+        // The total degree of coverage indicates how large the part of the sky vault is which is covered
+        // with clouds altogether. It is given in eighths, because this division is easier to estimate
+        // for the observer. 0 eighths means that there are no traces of clouds in the sky,
+        // 4 eighths means that the sky is covered with clouds up to half, 8 eighths means that the sky is
+        // completely covered with clouds and no sky blue can be recognized.
+        //
+        // For cloud determination, the observer will choose a location from which he can overlook the entire sky.
+        // When determining the total cloud cover, all clouds or parts of clouds, regardless of height and genus,
+        // that are above his location will be added. Aircraft contrails are also counted as cloud cover if they remain
+        // in the sky for more than 15 minutes. On the other hand, fog is not counted as cloud cover.
+        //
+        // With a little practice, the observer can easily determine the total amount of cloud cover.
+        // Problems occur only with scattered clouds or many single clouds (cumulus).
+        // Their cloud cover is often overestimated. On the other hand,
+        // the coverage of thin ice clouds (cirrus) is often underestimated [2].
+        //
+        // \par References:
+        // ISO 15469:2004(E) / CIE S 011/E:2003 : Spatial distribution of skylight
+        // https://en.wikipedia.org/wiki/Cloud_cover
+        // [1] https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html
+        // [2] https://www.dwd.de/DE/service/lexikon/Functions/glossar.html?lv2=100932&lv3=101016
+        //
+        enum CloudLayerLevel
+        {
+            // Cloud State is unknown (must not be used in ground truth).
+            //
+            CLOUD_LAYER_LEVEL_UNKNOWN = 0;
+
+            // Other (unspecified but known) CloudState.
+            //
+            CLOUD_LAYER_LEVEL_OTHER = 1;
+
+            // Cloudless (0/8)
+            //
+            CLOUD_LAYER_LEVEL_0 = 2;
+
+            // Sunny (1/8)
```

Comment:
Probably something less sunny should be used/added for night time coverage in the comment.

Reply:
Harmonization Group from 1/18/2022:
Changed to -> x/8 of the sky is covered with clouds.